File: isPalinDrome.js
const isNonAlphaNumeric = (char) => !/^[\da-z]$/i.test(char);

/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  let [left, right] = [0, s.length - 1];

  while (left < right) {
    while (left < right && isNonAlphaNumeric(s[left])) left++;
    while (left < right && isNonAlphaNumeric(s[right])) right--;

    if (s[left].toLowerCase() !== s[right].toLowerCase()) return false;

    left++;
    right--;
  }
  return true;
};

File: JavaWorkSpace/workspace/src/JavaPlayGround/IsPalindrome.java
package JavaPlayGround;

public class IsPalindrome {
    private boolean isNonAlphanumeric(char ch) {
        return !Character.isDigit(ch) && !Character.isLetter(ch);
    }

    public boolean isPalindrome(String s) {
        int left = 0, right = s.length() - 1;
        while (left < right) {

            while (isNonAlphanumeric(s.charAt(left)))
                left++;
            while (isNonAlphanumeric(s.charAt(right)))
                right--;

            if (Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right))) {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}
